File: 9. Palindrome Number/isPalindrome.js
/**
 * @param {number} x
 * @return {boolean}
 */
const isPalindrome = (x) => {
  if (x < 0) {
    return false;
  }

  let reversed = 0;
  let original = x;

  while (x > 0) {
    const digit = x % 10;
    reversed = reversed * 10 + digit;
    x = Math.floor(x / 10);
  }

  return reversed === original;
};


const x = 121;
console.log('isPalindrome: ', isPalindrome(x)); // true

File: 9. Palindrome Number/isPalindrome.py
class Solution:
  def isPalindrome(self, x: int) -> bool:
    if x < 0:
      return False
    
    reversed = 0
    original = x
    while x > 0:
      digit = x % 10
      reversed = reversed * 10 + digit
      x = x // 10

    return original == reversed
  

x = 121
solution = Solution()
print('isPalindrome: ', solution.isPalindrome(x)) # True
